### DISCUSSION TITLE
user html for static images?

### DISCUSSION THREAD

**Discussion by ppbrown** [Motion-Project/motionplus#66] [General]
For bandwidth reasons, I want the server to only serve up static snapshots of the camera view.
User will have to manually refresh page, to get updated image.
I've tried various settings in the conf files, but no luck.
Seems like the most straightforward way is to take advantage of 
"webcontrol_interface user"

But... what URL can I use in the html for this?
Even when I poke around directly with the default interface, and tell my browser "open image in new tab".... it uses a dynamic url, that keeps refreshing the image automatically :(

**Comment by Mr-Dave** [comment]
Not sure what you mean by dynamic.  A sample of the webcontrol html is in the "doc" subdirectory of the code [here](https://github.com/Motion-Project/motionplus/blob/master/doc/samplepage.html)  This sample is a bit outdated but can cycle through grabbing static images to refresh the preview page if needed and when users select camera specific it goes to the mjpeg stream.  

For just static images, the page could be really simple and just point to the static jpg urls.  All the information regarding the streams and uris is provided via the http://MotionplusUrl:port/0/config.json  Other options from the page are listed [here](https://motion-project.github.io/motionplus_examples.html#webcontrol_pages)

**Comment by ppbrown** [comment]
Thanks for the starting point.

related question: the nice url reference page you mention, says there are
urls to view  the main, and substream outputs, for the camera.
but.. I dont see anything in the config docs about how to specify the
streams for each of those, from the camera?

**Comment by Mr-Dave** [comment]
If I said that the `{camid}` shown in the url is the `device_id`  used in the configuration for each camera, is that what you are trying to determine?   (I've forgotten to keep these in sync with the terminology)

**Comment by ppbrown** [comment]
ah okay lemme be more specific.

Terminology first:
typically ,in IP cameras that support RTSP, you often have choice of a
"primary stream" and a 'sub stream".
For example, in reolink cameras,

Main Stream: ***@***.***_address:554/h264Preview_01_main

Sub Stream: ***@***.***_address:554/h264Preview_01_sub

primary, aka main stream, is full resolution.
sub stream is often a preview stream, at 640x480 or equivalent.

https://motion-project.github.io/motionplus_examples.html#webcontrol_pages
mentions

{IP}:{port0}/{camid}/static Primary image for the camera

{IP}:{port0}/{camid}/static/substream Substream image for the camera

In the sample camera-dist.conf , it mentions
netcam_url

in the docs, file:///C:/Users/phil/OneDrive/Documents/motionplus_config.html#netcam_url

it only says (give an RTSP stream)
it doesnt say anything about a sub stream!

But, now I guess I'll answer my own question, because now in the FULL
docs, I notice there is also a

netcam_high_url

Which is apparently intended for the full-res stream.

Seems like it could be a good thing to include that in the sample
camera configs.

**Comment by Mr-Dave** [comment]
Ah.  
The "main" stream that Motionplus will send is the one that you've specified as the `netcam_url`  That can be any one of the streams that the camera provides.

The url that is specified in the `netcam_high_url` will not be streamed out from MotionPlus.  It is for recording the stream without encoding.

The substream from MotionPlus will be created by MotionPlus by chopping in half the resolution from whatever is coming from `netcam_url`..subject to the result still having a permitted modulo set of dimensions.

All this is of course undocumented, or if so, in a very cryptic way.....
